### PR TITLE
Fix #684: AnalyzeZipArchiveEntry does not handle ArgumentException from ModulePath.FromFullPath

### LINKED ISSUE #684: AnalyzeZipArchiveEntry does not handle ArgumentException from ModulePath.FromFullPath
From WER:

```
Level . . . Symbol - InstructionPointer - FunctionOffset - SourceLine - SourceFile
0 . . . microsoft_pythontools_analysis!Microsoft.PythonTools.Analysis.ModulePath.FromFullPath - 0x00000000 - 0x000001F4 - - <<Failure related >>
1 . . . microsoft_pythontools!Microsoft.PythonTools.Intellisense.VsProjectAnalyzer.AnalyzeZipArchiveEntry - 0x00000000 - 0x000000D5 - - 
2 . . . microsoft_pythontools!Microsoft.PythonTools.Intellisense.VsProjectAnalyzer+__c__DisplayClass97_1._AnalyzeZipArchiveWorker_b__7 - 0x00000000 - 0x00000073 - - 
3 . . . microsoft_pythontools!Microsoft.PythonTools.Intellisense.ParseQueue+__c__DisplayClass5_0._EnqueueZipArchiveEntry_b__0 - 0x00000000 - 0x000000CC - - 
4 . . . microsoft_pythontools!Microsoft.PythonTools.Intellisense.ParseQueue+__c__DisplayClass6_0._EnqueWorker_b__0 - 0x00000000 - 0x00000025 - - 
5 . . . mscorlib_ni!System.Threading.QueueUserWorkItemCallback.WaitCallback_Context - 0x66EB0000 - 0x00000042 - - 
```

### PR DESCRIPTION
**Pull request by int19h**
Gracefully handle entry names inside zip archives that are not valid filenames, or do not correspond to a valid Python module name.

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs [hunk 1]
```diff
@@ -1420,7 +1420,12 @@ where packageDirs.Contains(path)
             }
             try {
                 string pathInZip = entry.FullName.Replace('/', '\\');
-                string path = Path.Combine(zipFileName, pathInZip);
+                string path;
+                try {
+                    path = Path.Combine(zipFileName, pathInZip);
```
Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs]: The CommonUtils equivalent here is better (if it works) - we'll be able to exceed 260 chars, which I believe actually works okay here.

File: Python/Tests/Core/PythonProjectTests.cs [hunk 2]
```diff
@@ -314,5 +314,19 @@ public class PythonProjectTests {
         }
 
 
+        [TestMethod, Priority(0)]
+        public void AnalyzeBadEgg() {
```
Comment (zooba) [inline Python/Tests/Core/PythonProjectTests.cs]: I really like this test name for some reason :)

File: Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs (hunk 1 above)
Comment (int19h) [inline Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs]: Seems to work fine.

File: Python/Tests/Core/PythonProjectTests.cs (hunk 2 above)
Comment (int19h) [inline Python/Tests/Core/PythonProjectTests.cs]: By design :)

**Comment by zooba** [comment]
+1

**Pull request merged**